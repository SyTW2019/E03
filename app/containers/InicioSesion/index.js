import React, { Component } from 'react';
import { BrowserRouter as Router, Switch, Route, Link } from 'react-router-dom';
import axios from 'axios';
import SignIn from 'components/SignIn';

export default function InicioSesion() {

  function nueva(user){
    axios.post('http://localhost:3000/usuario', user)
    .then(function (response) {
      console.log(response)
    })
    .catch(function (error) {
      console.log(error)
    });
   }

  return (
    <div>
      <SignIn action={nueva} />
    </div>
  );

}
