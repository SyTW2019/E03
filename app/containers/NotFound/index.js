import React from 'react';
import { Link } from 'react-router-dom';
import PageNotFound from './images/PageNotFound.jpg';

export default function NotFound() {
  
  return(
    <div>
    <img src={PageNotFound} style={{display: 'block', margin: 'auto', position: 'relative' }} alt="Error, pÃ¡gina no encontrada" />
    </div>
  );
}
