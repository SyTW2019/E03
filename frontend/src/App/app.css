
body {
	
background: rgb(174,229,238);
background: radial-gradient(circle, rgba(174,229,238,1) 0%, rgba(25,111,212,1) 100%);


/* otras opciones de colores abajo*/

/*background: rgb(174,238,213);
background: radial-gradient(circle, rgba(174,238,213,1) 0%, rgba(77,21,233,1) 83%);*/ 
 /*background: rgb(174,238,213);
background: radial-gradient(circle, rgba(174,238,213,1) 0%, rgba(2,2,64,1) 83%); */ 

 /*background: rgb(207,111,134);
background: radial-gradient(circle, rgba(207,111,134,1) 0%, rgba(212,21,52,1) 8%, rgba(119,65,66,1) 16%, rgba(138,158,31,1) 27%, rgba(138,174,71,1) 39%, rgba(51,201,49,1) 46%, rgba(78,168,166,1) 59%, rgba(90,51,121,1) 65%, rgba(5,5,45,1) 86%); */

}

/* Main content */
.main {
    margin-top: 30px; 
} /*Add a top margin to avoid content overlay */

.try {
       
	
	margin-top: 120px;
	margin-bottom: 200px;

	
}

.head{
	background-color: white;
	margin-top: -1%;
	
}
  

