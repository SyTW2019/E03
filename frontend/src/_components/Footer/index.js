import React from 'react'
import { FormattedMessage } from 'react-intl'

import Wrapper from './Wrapper'
import messages from './messages'

function Footer() {
  return (
    <div class="d-flex border-top mt-4 align-items-center justify-content-around bg-light">
      <div class="bd-highlight mt-3 mb-5">
        <div class="mb-1 text-secondary">EVENTSFINDER</div>

        <div>
          Esta página web le permite crear eventos, apuntarse a ellos, <br />
          compartirlos, así como ver los eventos próximos y <br />
          acceder a una extensa comunidad de eventos.
        </div>
      </div>

      <div class="bd-highlight mt-3 mb-5 pb-4 mr-5 pr-5">
        <div class="mb-1 text-secondary">ENLACE AL PROYECTO</div>

        <div class="d-flex align-items-center">
          <img
            src="../../../images/github.png"
            alt="logoGitHub"
            class="img-responsive img-fluid"
            height="30"
            width="30"
          />

          <a href="https://github.com/SyTW2019/E03" class="ml-1">
            GitHub del Proyecto
          </a>
        </div>
      </div>

      <div class="bd-highlight mt-3 mb-2">
        <div class="mb-1 text-secondary">CONTACTO</div>

        <div class="d-flex align-items-center">
          <img
            src="../../../images/correo.jpg"
            alt="logoCorreo"
            class="img-responsive img-fluid ml-1"
            height="15"
            width="15"
          />

          <a href="alu0101061672@ull.edu.es" class="pl-2 small">
            alu0101061672@ull.edu.es
          </a>
        </div>

        <div class="d-flex align-items-center">
          <img
            src="../../../images/correo.jpg"
            alt="logoCorreo"
            class="img-responsive img-fluid ml-1"
            height="15"
            width="15"
          />

          <a href="alu0101029993@ull.edu.es" class="pl-2 small">
            alu0101029993@ull.edu.es
          </a>
        </div>

        <div class="d-flex align-items-center">
          <img
            src="../../../images/correo.jpg"
            alt="logoCorreo"
            class="img-responsive img-fluid ml-1"
            height="15"
            width="15"
          />

          <a href="alu0100954609@ull.edu.es" class="pl-2 small">
            alu0100954609@ull.edu.es
          </a>
        </div>

        <div class="d-flex align-items-center">
          <img
            src="../../../images/correo.jpg"
            alt="logoCorreo"
            class="img-responsive img-fluid ml-1"
            height="15"
            width="15"
          />

          <a href="alu0101028026@ull.edu.es" class="pl-2 small">
            alu0101028026@ull.edu.es
          </a>
        </div>

        <div class="d-flex align-items-center">
          <img
            src="../../../images/correo.jpg"
            alt="logoCorreo"
            class="img-responsive img-fluid ml-1"
            height="15"
            width="15"
          />

          <a href="alu0101045603@ull.edu.es" class="pl-2 small">
            alu0101045603@ull.edu.es
          </a>
        </div>
      </div>
    </div>
    // <Wrapper>

    // <section>
    //     <FormattedMessage {...messages.message} />
    // </section>

    // </Wrapper>
  )
}

export default Footer
