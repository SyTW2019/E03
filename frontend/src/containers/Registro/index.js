import React, { Component } from 'react';
import { BrowserRouter as Router, Switch, Route, Link } from 'react-router-dom';
import axios from 'axios';
import SignUp from 'components/SignUp';

export default function Registro() {

  function nueva(user){
    axios.post('http://localhost:3000/usuario', user)
    .then(function (response) {
      console.log(response)
    })
    .catch(function (error) {
      console.log(error)
    });
   }

  return (
    <div>
      <SignUp action={nueva} />
    </div>
  );

}
