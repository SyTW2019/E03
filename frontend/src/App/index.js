import React from 'react';
import { Router, Route, Switch, Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import {IntlProvider} from 'react-intl';
import { history } from '../_helpers';
import { alertActions } from '../_actions';
import { PrivateRoute } from '../_components/PrivateRoute';
import { HomePage } from '../HomePage';
import { LoginPage } from '../LoginPage';
import { RegisterPage } from '../RegisterPage';
import { PaginaPrincipal } from '../PaginaPrincipal';
import Categoria from '../Categorias';
import Header from '../_components/Header'
import Footer from '../_components/Footer'

class App extends React.Component {
    constructor(props) {
        super(props);

        history.listen((location, action) => {
            // clear alert on location change
            this.props.clearAlerts();
        });
    }

    render() {
        const { alert } = this.props;
        return (
            <IntlProvider locale="en">
                <div>
                    {alert.message &&
                        <div className={`alert ${alert.type}`}>{alert.message}</div>
                    }
                    <Router history={history}>
                    <div> 
                        <Header />
                        <Switch>
                            <Route exact path="/" component={PaginaPrincipal} />
                            <Route path="/login" component={LoginPage} />
                            <Route path="/register" component={RegisterPage} />
                            <Route path="/categoria" component={Categoria} />
                            <PrivateRoute exact path="/home_user" component={HomePage} />
                            <Redirect from="*" to="/" />
                        </Switch>
                        
                    </div>
                    </Router>
                    <div class="fixed-bottom mb-3">
                        <Footer />
                    </div>
                </div>
            </IntlProvider>
        );
    }
}

function mapState(state) {
    const { alert } = state;
    return { alert };
}

const actionCreators = {
    clearAlerts: alertActions.clear
};

const connectedApp = connect(mapState, actionCreators)(App);
export { connectedApp as App };

