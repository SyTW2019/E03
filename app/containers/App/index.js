/**
 *
 * App
 *
 * This component is the skeleton around the actual pages, and should only
 * contain code that should be seen on all pages. (e.g. navigation bar)
 */

import React, { Component } from 'react';
import { Helmet } from 'react-helmet';
import styled from 'styled-components';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';

import axios from 'axios';

import Footer from 'components/Footer'
import Header from 'components/Header'

import GlobalStyle from '../../global-styles';

import Registro from '../Registro';
import Inicio from '../PaginaPrincipal';
import InicioSesion from '../InicioSesion';
import Categoria from '../Categorias';
import NotFound from '../NotFound';

class App extends Component {
  render() {
    return (
      <IntlProvider locale="en">
        <Router>
          <div>
            <Header />
            <Switch>
              <Route exact path="/" component={Inicio} />
              <Route path="/SignUp" component={Registro} />
              <Route path="/SignIn" component={InicioSesion} />
              <Route path="/Category" component={Categoria} />
              <Route component={NotFound} />
            </Switch>
            <Footer />
          </div>
        </Router>
      </IntlProvider>
      /*   <IntlProvider locale="en">
           <Router>
             <Header/>
             <Switch>
               <Route exact path='/' Component={Card}><Card/></Route>
               <Route path='/SignUp' Component={Registro}><Registro/></Route>
               <Route path='/SignIn' Component={Inicio}><Inicio/></Route>
               <Route path='/Category' Component={Categoria}><Categoria/></Route>
               <Route path="*" component={NotFound}><NotFound/></Route>
             </Switch>
             <Footer/>
           </Router>
         </IntlProvider>*/

    );
  }
}

export default App;
