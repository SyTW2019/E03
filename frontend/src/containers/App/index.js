/**
 *
 * App
 *
 * This component is the skeleton around the actual pages, and should only
 * contain code that should be seen on all pages. (e.g. navigation bar)
 */

import React from 'react';
import { IntlProvider } from 'react-intl';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';

import Footer from 'components/Footer'
import Header from 'components/Header'
import Registro from '../Registro';
import Inicio from '../PaginaPrincipal';
import InicioSesion from '../InicioSesion';
import Categoria from '../Categorias';
import NotFound from '../NotFound';

import { HomePage } from '../HomePage';
import { LoginPage } from '../LoginPage';
import { RegisterPage } from '../RegisterPage';
import { history } from '../../components/_helpers';
import { alertActions } from '../../components/_actions';
import { PrivateRoute } from '../../components/_components';

class App extends React.Component {
  render() {
    return (
      <IntlProvider locale="en">
        <Router>
          <div>
            <Header />
            <Switch>
              <Route exact path="/" component={Inicio} />
              <Route path="/SignUp" component={Registro} />
              <Route path="/SignIn" component={InicioSesion} />
              <Route path="/Category" component={Categoria} />
              <Route component={NotFound} />
            </Switch>
            <Footer />
          </div>
        </Router>
      </IntlProvider>
      /*   <IntlProvider locale="en">
           <Router>
             <Header/>
             <Switch>
               <Route exact path='/' Component={Card}><Card/></Route>
               <Route path='/SignUp' Component={Registro}><Registro/></Route>
               <Route path='/SignIn' Component={Inicio}><Inicio/></Route>
               <Route path='/Category' Component={Categoria}><Categoria/></Route>
               <Route path="*" component={NotFound}><NotFound/></Route>
             </Switch>
             <Footer/>
           </Router>
         </IntlProvider>*/

    );
  }
}

export default App;
