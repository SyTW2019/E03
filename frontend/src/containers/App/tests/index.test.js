import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import ReactDOM from 'react-dom';
import { render, cleanup } from 'react-testing-library';
import "jest-dom/extend-expect";
import Registro from '../../Registro'
import Inicio from '../../PaginaPrincipal';
import InicioSesion from '../../InicioSesion';
import Categoria from '../../Categorias';
import NotFound from '../../NotFound';
import Footer from '../../../components/Footer';

import App from '../index';

const renderer = new ShallowRenderer();

describe('<App />', () => {

  it('render', () => {
    const div = document.createElement(div);
    ReactDOM.render(<App/>,div);
  });

  it('should render and match the snapshot', () => {
    renderer.render(<App />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });

  it('should render an Card component', () => {
    const { container } = render(<Inicio />);
    expect(container).not.toBeNull();
  });

  it('should render an Registro component', () => {
    const { container } = render(<Registro />);
    expect(container).not.toBeNull();
  });

  // it('should render an Inicio component', () => {
  //   const { container } = render(<InicioSesion />);
  //   expect(container).not.toBeNull();
  // });

  it('should render an Categoria component', () => {
    const { container } = render(<Categoria />);
    expect(container).not.toBeNull();
  });

  it('should render an NotFound component', () => {
    const { container } = render(<NotFound />);
    expect(container).not.toBeNull();
  });

});
