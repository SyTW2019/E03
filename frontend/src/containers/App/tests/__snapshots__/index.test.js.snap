// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render and match the snapshot 1`] = `
<IntlProvider
  locale="en"
>
  <BrowserRouter>
    <div>
      <Header />
      <Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/"
        />
        <Route
          component={[Function]}
          path="/SignUp"
        />
        <Route
          component={[Function]}
          path="/SignIn"
        />
        <Route
          component={[Function]}
          path="/Category"
        />
        <Route
          component={[Function]}
        />
      </Switch>
      <Footer />
    </div>
  </BrowserRouter>
</IntlProvider>
`;
